### PR TITLE
godaddy.com suffix+similar_domain features

### PR DESCRIPTION
**Pull request by madalin-dogaru**
Added: 
- alternative suffix search on godaddy.com
- similarly looking domains available on godaddy.com

### CODE REVIEW

File: godaddy_search.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import itertools
 import requests
+from termcolor import colored
 
-def generate_similar_domains(domain, num_domains=3):
+def generate_similar_domains(domain, num_domains=10):
     words = domain.split(".")[0].split("-")
-    tlds = [".com", ".net", ".org", ".io", ".co"]
+    tlds = [".com", ".net", ".org", ".io", ".co", ".ai", ".us", ".co", ".de"]
```
Comment (teodor27) [inline godaddy_search.py]: Why ".co" again ?

File: godaddy_search.py [hunk 2]
```diff
@@ -34,12 +35,39 @@ def check_domain_availability(domain):
     else:
         return None
 
-def query_similar_domains(domain):
-    similar_domains = generate_similar_domains(domain)
+def replace_domain_characters(domain, replacements):
+    replaced_domains = []
+    domain_name, tld = domain.rsplit(".", 1)
```
Comment (teodor27) [inline godaddy_search.py:40]: Maybe save dot character "." to constant

File: README.md [hunk 3]
```diff
@@ -22,23 +22,29 @@ Examples
...
 
+3. Supply a domain and get other available suffixes on goddady.com (requires API token).   
+`python3 main.py -daddy microsoft.com`   
+
+4. Supply a domain and get similarly looking domains for that domain and suffix that are available on godaddy.com. (requires API token)   
+`main.py -domphish microsoft.com`
+
 3. Take a list of IP's, get their Country/City/Area and write it in a file(including the IPs).    
-`python3 main.py -iplist file_containing_ips -o output_file_name`
+`python3 main.py -iplist file_containing_ips -o output_file_name`
```
Comment (teodor27) [inline README.md]: double check unnecessary space characters

File: __pycache__/godaddy_search.cpython-311.pyc
Comment (teodor27) [inline __pycache__/godaddy_search.cpython-311.pyc:1]: I suggest to configure a .gitignore to avoid pushing such files

File: godaddy_search.py [hunk 4]
```diff
@@ -34,12 +35,39 @@ def check_domain_availability(domain):
...
+    return replaced_domains
 
-    for domain in similar_domains:
-        is_available = check_domain_availability(domain)
-        if is_available is None:
-            print(f"Error occurred while checking {domain}")
-        else:
-            print(f"{domain} is {'available' if is_available else 'not available'}")
+
+def query_similar_domains(domain, replacements=None):
```
Comment (teodor27) [inline godaddy_search.py]: double check code formatting  for replacements = None

**Review by teodor27 — CHANGES_REQUESTED** [review]

File: godaddy_search.py (hunk 1 above)
Comment (madalin-dogaru) [reply godaddy_search.py]: fixed

File: README.md (hunk 3 above)
Comment (madalin-dogaru) [reply README.md]: fixed

File: godaddy_search.py (hunk 4 above)
Comment (madalin-dogaru) [reply godaddy_search.py]: improved readability for multiple symbols in all the *.py files.

File: godaddy_search.py (hunk 2 above)
Comment (madalin-dogaru) [reply godaddy_search.py:40]: will take in consideration in future versions once prototyping is completed.

File: __pycache__/godaddy_search.cpython-311.pyc
Comment (madalin-dogaru) [reply __pycache__/godaddy_search.cpython-311.pyc:1]: added to .gitignore: 
```
# Compilation Trash #   
__pycache__/   
*.pyc   
*.pyo   
*.pyd
```

**Pull request merged**